ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -989,3 +989,12 @@ let with_vm_operation ~__context ~self ~doc ~op ?(strict=true) ?policy f =
          Helpers.Early_wakeup.broadcast (Datamodel._vm, Ref.string_of self);
        with
          _ -> ())
+
```

Comment:
I have to look into this in context but this code can be improved by avoiding list concatenation:

```
let ensure_device_model_profile_present ~__context ~hVM_boot_policy platform =
  let is_hvm = hVM_boot_policy <> "" in
  let default = Vm_platform.(device_model, default_device_model_default_value) in
  if not is_hvm || List.mem_assoc Vm_platform.device_model platform then
    platform
  else
    default :: platform
```
I find the name `with_device_model_profile` is a bit unfortunate. Functions named `with_..` are often typically higher-order functions that take a function as an argument - this does not. What this function does is the guarantee that the device model profile is defined. I'm not convinced that my name is much better, though.

Reply:
Thanks, updated.